### PR TITLE
Prep for Registry FFI; Convert RegDeleteKeyEx to FFI

### PR DESCRIPTION
**Pull request by chefsalim**
Enable Registry to use FFI ; convert delete_key_ex to use FFI version of RegDeleteKeyEx API.

### CODE REVIEW

File: lib/chef/win32/api/registry.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#
+# Author:: Salim Alam (<salam@chef.io>)
+# Copyright:: Copyright 2015 Opscode, Inc.
```
Comment (stevendanna) [inline lib/chef/win32/api/registry.rb]: s/Opscode/Chef Software/

File: lib/chef/win32/registry.rb [hunk 2]
```diff
@@ -142,9 +153,13 @@ def delete_key(key_path, recursive)
       #Using the 'RegDeleteKeyEx' Windows API that correctly supports WOW64 systems (Win2003)
       #instead of the 'RegDeleteKey'
       def delete_key_ex(hive, key)
-        regDeleteKeyEx = ::Win32::API.new('RegDeleteKeyEx', 'LPLL', 'L', 'advapi32')
         hive_num = hive.hkey - (1 << 32)
-        regDeleteKeyEx.call(hive_num, key, ::Win32::Registry::KEY_WRITE | registry_system_architecture, 0)
+        begin
+          RegDeleteKeyExW(hive_num, wstring(key), ::Win32::Registry::KEY_WRITE | registry_system_architecture, 0)
+          return true
```
Comment (stevendanna) [inline lib/chef/win32/registry.rb]: Minor style point: We'd typically elide the `return` keyword here and below.

**Comment by stevendanna** [comment]
Left a couple minor style comments.  Otherwise LGTM.

File: lib/chef/win32/registry.rb [hunk 3]
```diff
@@ -142,9 +153,13 @@ def delete_key(key_path, recursive)
...
+        rescue ::Win32::Registry::Error => e
```
Comment (jaym) [inline lib/chef/win32/registry.rb]: Where is this error coming from? ffi won't throw this

**Comment by chefsalim** [comment]
@stevendanna @jaym Good points - fixed.

**Comment by jaym** [comment]
:+1:

**Comment by stevendanna** [comment]
:+1: lgtm

**Pull request merged**